scripts/partition_manager/partition_manager.rst

```diff
@@ -49,6 +49,27 @@ This file must be stored in the same folder as the main :file:`CMakeLists.txt` f
    The root application does not need to define a :file:`pm.yml` file, because its partition size and placement is implied by the size and placement of the sub-image partitions.
    If a root application defines a :file:`pm.yml` file, it is silently ignored.
 
+A subsystem can also provide Partition Manager configuration.
+This is done by adding the path to the :file:`pm.yml` file to the global property ``PM_SUBSYS``.
+Subsystem Partition Manager configurations can not define image partitions.
+The listing below shows how to properly define the Partition Manager configuration for a subsystem.
+This could would be placed in the :file:`CMakeLists.txt` of that subsystem.
+
+.. code-block:: cmake
+
+   if (NOT IMAGE)
```

Comment:
I'm wondering if we should find another approach that doesn't require this

    if (NOT IMAGE)

I think this will cause problems with the use case. I'm guessing we want to use it like this:

    if (FOO_ENABLED)
      ...
      set_property(
        GLOBAL APPEND PROPERTY
        PM_SUBSYS
        ${pm_path}
      )
    endif ()

So if FOO_ENABLED is defined in a sub-image, we cannot have `if (NOT IMAGE)`, but we still want FOO_ENABLED to be set in the sub-image via Kconfig (possibly via `select`) since that's the standard way to enable modules.

My suggestion:
Allow several sub-images to each define subsys partitions, but give an error if the definitions are not identical.

Reply:
Got it, so we still have the property that partitions are global, as opposed to per image, but we don't need this check?

Reply:
Exactly

Reply:
Fixed, PM_SUBSYS is now IMAGE-ified, and the python script will throw a runtime error if the configuration of a given partition differs.